src/Microsoft.Framework.PackageManager/NuGetConfig.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet;
+
+namespace Microsoft.Framework.PackageManager
+{
+    /// <summary>
+    /// Manages the NuGet.config hierarchy
+    /// </summary>
+    internal class NuGetConfig
+    {
+        public NuGetConfig(ISettings settings, PackageSourceProvider sources)
+        {
+            Settings = settings;
+            Sources = sources;
+        }
+
+        public ISettings Settings { get; }
+        public IPackageSourceProvider Sources { get; }
+
+        public static NuGetConfig ForSolution(string solutionDirectory)
+        {
+            return ForSolution(solutionDirectory, new PhysicalFileSystem(Directory.GetCurrentDirectory()));
+        }
+
+        public static NuGetConfig ForSolution(string solutionDirectory, IFileSystem fileSystem)
+        {
+            var settings = SettingsUtils.ReadSettings(
+                solutionDirectory, 
+                nugetConfigFile: null, 
+                fileSystem: fileSystem, 
+                machineWideSettings: new CommandLineMachineWideSettings());
+
+            // Recreate the source provider and credential provider
+            var sources = PackageSourceBuilder.CreateSourceProvider(settings);
+            //HttpClient.DefaultCredentialProvider = new SettingsCredentialProvider(new ConsoleCredentialProvider(Console), SourceProvider, Console);
```

Comment:
Sure, it's just moved from the old code so I wasn't sure if it was being left in for a reason.